### PR TITLE
ci: add .NET release CI

### LINKED ISSUE #95: .NET release flow is required
To migrate to .NET version plugin, we need a release tool to do cross-complie, create checksum file, update artifacts and create release.

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Feature:
- added release script for .NET project
- update CI to trigger release script

Example release: https://github.com/JeyJeyGao/notation-azure-kv/releases/tag/untagged-f7b59db6549c8908c202

Resolves #95 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    fi
+
+    (cd $artifacts_dir && sha256sum $(basename $artifact_name) >>${checksum_name})
+
+    # Add the artifact to the list
+    artifacts+=($artifact_name)
+done
+
+# Create a pre-release using GitHub CLI
+gh release create --title "Release $tag_name" --prerelease $tag_name ${artifacts[@]} ${checksum_name}
```
Comment (shizhMSFT) [inline scripts/release.sh:60]: Title should just be the tag name. We don't need to make it a `prerelease` but need to make it a draft.

File: -
Comment (shizhMSFT) [inline -:1]: What is this binary file `-`?

**Review by shizhMSFT — CHANGES_REQUESTED** [review]
Do you have a sample release page in your personal repo?

File: scripts/release.sh (hunk 1 above)
Comment (JeyJeyGao) [reply scripts/release.sh]: Updated.

File: -
Comment (JeyJeyGao) [reply -:1]: Removed.

**Review by shizhMSFT — COMMENTED** [review]
```console
$ notation plugin ls
NAME       DESCRIPTION                       VERSION   CAPABILITIES                ERROR
azure-kv   Notation Azure Key Vault plugin   1.0.0     [SIGNATURE_GENERATOR.RAW]   <nil>
```
I found that the `VERSION` is not the tag version. Are you able to change version in the linker at the build time based on the tag name?

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    fi
+
+    (cd "${artifacts_dir}" && sha256sum "$(basename "${artifact_name}")" >>"${checksum_name}")
+
+    # Add the artifact to the list
+    artifacts+=("${artifact_name}")
+done
+
+# Create a pre-release using GitHub CLI
+gh release create --title "${tag_name}" --draft "${tag_name}" "${artifacts[@]}" "${checksum_name}"
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply scripts/release.sh]: Updated.

**Comment by JeyJeyGao** [comment]
> I found that the `VERSION` is not the tag version. Are you able to change version in the linker at the build time based on the tag name?

Code updated. The version will be the tag name now.

File: scripts/release.sh [hunk 3]
```diff
@@ -59,5 +64,11 @@ for runtime in "${runtimes[@]}"; do
     artifacts+=("${artifact_name}")
 done
 
-# Create a pre-release using GitHub CLI
-gh release create --title "${tag_name}" --draft "${tag_name}" "${artifacts[@]}" "${checksum_name}"
+# Create a release using GitHub CLI
+if [[ "$tag_name" == "-" ]]; then
```
Comment (shizhMSFT) [inline scripts/release.sh:68]: `$ [[ "v1.0.0-rc.1" == "-" ]] && echo yes` prints nothing. It seems the condition expression is not correct.

File: scripts/release.sh [hunk 4]
```diff
@@ -59,5 +64,11 @@ for runtime in "${runtimes[@]}"; do
...
+    # v1.0.0-rc.1 is a pre-release
+    gh release create --title "${tag_name}" --prerelease "${tag_name}" "${artifacts[@]}" "${checksum_name}"
```
Comment (shizhMSFT) [inline scripts/release.sh:70]: We still need to make it a draft.
```suggestion
    gh release create --title "${tag_name}" --draft --prerelease "${tag_name}" "${artifacts[@]}" "${checksum_name}"
```

File: Notation.Plugin.AzureKeyVault/Command/GetPluginMetadata.cs [hunk 5]
```diff
@@ -5,9 +5,12 @@ namespace Notation.Plugin.AzureKeyVault.Command
     /// <summary>
     /// Implementation of get-plugin-metadata command.
     /// </summary>
-    public class GetPluginMetadata : IPluginCommand
+    public partial class GetPluginMetadata : IPluginCommand
     {
-        public const string Version = "1.0.0";
+        public static readonly string Version = "Unknown";
+        public static readonly string CommitTime = "Unknown";
+        public static readonly string CommitHash = "Unknown";
```
Comment (shizhMSFT) [inline Notation.Plugin.AzureKeyVault/Command/GetPluginMetadata.cs:12]: nit:
```suggestion
        public static readonly string Version = "1.0.0+unreleased";
        public static readonly string CommitTime = "";
        public static readonly string CommitHash = "";
```

File: Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj [hunk 6]
```diff
@@ -7,6 +7,11 @@
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <AssemblyName>notation-azure-kv</AssemblyName>
+
+    <!-- Build metadata -->
+    <Version Condition="'$(Version)' == ''">$(Version)</Version>
+    <CommitTime Condition="'$(CommitTime)' != ''">$(CommitTime)</CommitTime>
```
Comment (shizhMSFT) [inline Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj:13]: It is not necessary to have commit time

File: scripts/release.sh (hunk 4 above)
Comment (JeyJeyGao) [reply scripts/release.sh]: Updated.

File: Notation.Plugin.AzureKeyVault/Command/GetPluginMetadata.cs (hunk 5 above)
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Command/GetPluginMetadata.cs]: Updated. The default value is set in `.csproj`

File: Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj (hunk 6 above)
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj]: Removed.

File: Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj [hunk 7]
```diff
@@ -7,6 +7,11 @@
...
+    <CommitHash Condition="'$(CommitHash)' != ''">$(CommitHash)</CommitHash>
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj]: Updated. The default value in `.cspoj`  will be used:
>     <Version Condition="'$(Version)' != ''">$(Version)</Version>
>     <CommitHash Condition="'$(CommitHash)' != ''">$(CommitHash)</CommitHash>
>     <Version Condition="'$(Version)' == ''">1.0.0+unreleased</Version>
>     <CommitHash Condition="'$(CommitHash)' == ''"></CommitHash>

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**